### DISCUSSION TITLE
Converting existing repo to LFS for all branches/tags in repository without each one checkout

### DISCUSSION THREAD

**Discussion by rgaduput** [git-lfs/git-lfs#5344] [Q&A]
HI,
We have an on prem Bitbucket repository whose size more than 15G and does not currently contain any LFS configuration. So everything is stored at the repository level only.
We are in a process of migrating our on-prem to cloud Bitbucket, since cloud vendor does not allow repo size to be more than 2G (4GB), we have tried to clean the LFS objects from history to reduce the size using BFG tool and the run `git lfs install` to update the LFS hooks.
If run the `git push --all` LFS objects gets pushed into Bitbucket cloud, but we do not want this to happen. Since LFS size is huge we prefer to store them on on-prem LFS server (artifactory).
I know that there is way to LFS config using `git config`, but will we be able to run this on git bare repository ?
Will this push LFS objects for all branches and tags ? (we have 100's of branches and tags so its problematic to checkout all and add LFS config it to them)

Below are the steps followed 
1. `git clone --mirror <repository-url>`
2. Run `bfg` to clean the repository for all branches and tags `bfg --convert-to-git-lfs "*.{png,mp4}" --no-blob-protection <repo-name>.git`
3. `cd <repo-name>.git`
4. `git lfs install`
5. change remote url to cloud one (`git remote set-url origin <cloud_repo_url>`)
6. **Expectation** to be able to set LFS config for all branches/tags here before pushing.
7. `git push --force`

Have any one tried such before ? Appreciate any suggestion on how to achieve this.

Thanks in advance.

**Comment by bk2204** [comment]
Hey,

There are two options here.  One is to set the `lfs.url` configuration in `.git/config`, which will apply to all branches and configure the destination LFS server.  However, it's not stored with the repo, so everyone has to set it in order to push.

The other option is to add a Git-style config file as `.lfsconfig` in the root of the repository that configures `lfs.url`, which will apply to anyone who uses the repository unless they've explicitly overridden it, but it requires changing every branch.  That's not easy to do with Git without checking out each branch (although it's theoretically possible with a separate index and some plumbing), but you could use libgit2 to add an appropriate commit to every branch with just a bare repository.
